libs/ui/elements/src/lib/record-api-form/record-api-form.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core'
+import { DatasetServiceDistribution } from '@geonetwork-ui/common/domain/record'
+import { BehaviorSubject, combineLatest, map } from 'rxjs'
+
+const DEFAULT_PARAMS = {
+  OFFSET: '0',
+  LIMIT: '20',
+  FORMAT: 'json',
+}
+@Component({
+  selector: 'gn-ui-record-api-form',
+  templateUrl: './record-api-form.component.html',
+  styleUrls: ['./record-api-form.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RecordApiFormComponent {
+  @Input() set apiLink(value: DatasetServiceDistribution) {
+    this.linkHref = new URL('/data/swagger-ui/index.html', value.url.origin)
+    this.apiBaseUrl = value.url.href
+    this.resetUrl()
+  }
+  offset$ = new BehaviorSubject('')
```

Comment:
Why do you use Observables everywhere ?
Classic change detection is meant for this.
BTW, we could take the opportunity to test signals !

Reply:
Actually, the other solutions we tried to get the different inputs to react to changes made from the code (eg : reset to the initial values), were not working with classic change detection, hence the Observables. But let me know if you have other suggestions, happy to hear them.

Reply:
Did you try `ChangeDetectorRef` ?
Actually I don't know what is the best practice about that.

Reply:
I believe we did, right @tkohr ?

Reply:
Indeed, we had some change detection problems when switching cards which we were not able to resolve (even by manually calling `ChangeDetectorRef.detectChanges()`)